### ISSUE TITLE
Error when trying to run MobileNetV2 on esp32 (abort)

### ISSUE THREAD

**Issue by YarivCol** [tensorflow/tensorflow#42170]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04 (tensorflow), Windows 10 (microcontroller ide)
- TensorFlow installed from (source or binary): 
- Tensorflow version (commit SHA if source): 2.3.0
- Target platform (e.g. Arm Mbed OS, Arduino Nano 33 etc.): esp32(with external ram)

**Describe the problem**

I am trying to run MobileNetV2 on esp32, the model creation and conversion code is:

`base_model = tf.keras.applications.MobileNetV2(input_shape=(48, 48, 1), alpha=0.35, weights=None, include_top=False)
x = base_model.output
x = tf.keras.layers.Flatten()(x)
x = tf.keras.layers.Dense(2)(x) #final layer with softmax activation for N classes
preds = tf.keras.layers.Softmax()(x)
model = tf.keras.models.Model(inputs=base_model.input,outputs=preds) #specify the inputs and outputs
converter = tf.lite.TFLiteConverter.from_keras_model(model)
def representative_dataset():
  for i in range(500):
    yield([np.random.rand(1,48,48,1).astype(np.float32)])
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
converter.representative_dataset = representative_dataset
tflite_model = converter.convert()
open("mobilenet_test.tflite", "wb").write(tflite_model)`

I am loading the tflite into partition called tensorflow_model, the esp32 code:
`    // alloc the tensor data on external RAM
    tensor_arena = (uint8_t *) heap_caps_malloc(kTensorArenaSize, MALLOC_CAP_SPIRAM);

    printf("tesnor arena at addr: %p\r\n", tensor_arena);

    // Find the cry_model partition in the partition table
    const esp_partition_t *partition = esp_partition_find_first(ESP_PARTITION_TYPE_DATA, 
                                                                ESP_PARTITION_SUBTYPE_ANY, 
                                                                "tensorflow_model");

    // Map the partition to data memory
    ESP_ERROR_CHECK(esp_partition_mmap(partition, 
                                       0, 
                                       partition->size, 
                                       SPI_FLASH_MMAP_DATA, 
                                       &tensorflow_model_data,
                                       &map_handle));

    ESP_LOGI(TAG, "Mapped tensorflow_model partition to data memory address %p", tensorflow_model_data);

    // Set up logging. Google style is to avoid globals or statics because of
    // lifetime uncertainty, but since this has a trivial destructor it's okay.
    // NOLINTNEXTLINE(runtime-global-variables)
    static tflite::MicroErrorReporter micro_error_reporter;
    error_reporter = &micro_error_reporter;

    // Map the model into a usable data structure. This doesn't involve any
    // copying or parsing, it's a very lightweight operation.
    model = tflite::GetModel(tensorflow_model_data);
    if (model->version() != TFLITE_SCHEMA_VERSION) 
    {
        TF_LITE_REPORT_ERROR(error_reporter,
                             "Model provided is schema version %d not equal "
                             "to supported version %d.",
                             model->version(), TFLITE_SCHEMA_VERSION);
        return;
    }

    static tflite::AllOpsResolver resolver;

    // Build an interpreter to run the model with.
    static tflite::MicroInterpreter static_interpreter(model,
                                                       resolver, 
                                                       tensor_arena, 
                                                       kTensorArenaSize, 
                                                       error_reporter);
    interpreter = &static_interpreter;

    // Allocate memory from the tensor_arena for the model's tensors.
    TfLiteStatus allocate_status = interpreter->AllocateTensors();
    if (allocate_status != kTfLiteOk) 
    {
        ESP_LOGI(TAG, "AllocateTensors() failed");
        return;
    }
    TfLiteStatus invoke_status = interpreter->Invoke();
    if (invoke_status != kTfLiteOk) {
      TF_LITE_REPORT_ERROR(error_reporter, "Invoke failed\n");
    }`

the code crush in invoke with :
abort() was called at PC 0x401134f7 on core 0
0x401134f7: tflite::PreprocessSoftmaxScaling(double, double, int, int*, int*) at C:\work\projects\rhino\build/../components/tfmicro/tensorflow/lite/kernels/internal/quantization_util.cc:96
 (inlined by) tflite::PreprocessSoftmaxScaling(double, double, int, int*, int*) at C:\work\projects\rhino\build/../components/tfmicro/tensorflow/lite/kernels/internal/quantization_util.cc:296

ELF file SHA256: 8bf88518f1fe95d5

Backtrace: 0x40087ca9:0x3ffb5390 0x40088061:0x3ffb53b0 0x401134f7:0x3ffb53d0 0x400ed948:0x3ffb5400 0x400d93c2:0x3ffb5480 0x400d548d:0x3ffb54a0 0x400d562f:0x3ffb54d0 0x400d0ff0:0x3ffb5610 0x4008c141:0x3ffb5630
0x40087ca9: invoke_abort at C:/work/esp-idf/components/esp32/panic.c:155

0x40088061: abort at C:/work/esp-idf/components/esp32/panic.c:172

0x401134f7: tflite::PreprocessSoftmaxScaling(double, double, int, int*, int*) at C:\work\projects\rhino\build/../components/tfmicro/tensorflow/lite/kernels/internal/quantization_util.cc:96
 (inlined by) tflite::PreprocessSoftmaxScaling(double, double, int, int*, int*) at C:\work\projects\rhino\build/../components/tfmicro/tensorflow/lite/kernels/internal/quantization_util.cc:296

0x400ed948: tflite::ops::micro::activations::SoftmaxEval(TfLiteContext*, TfLiteNode*) at C:\work\projects\rhino\build/../components/tfmicro/tensorflow/lite/micro/kernels/softmax.cc:57
 (inlined by) tflite::ops::micro::activations::SoftmaxEval(TfLiteContext*, TfLiteNode*) at C:\work\projects\rhino\build/../components/tfmicro/tensorflow/lite/micro/kernels/softmax.cc:118

0x400d93c2: tflite::MicroInterpreter::Invoke() at C:\work\projects\rhino\build/../components/tfmicro/tensorflow/lite/micro/micro_interpreter.cc:285

0x400d548d: setup_model_data() at C:\work\projects\rhino\build/../main/main.cc:145

0x400d562f: app_main at C:\work\projects\rhino\build/../main/main.cc:232

0x400d0ff0: main_task at C:/work/esp-idf/components/esp32/cpu_start.c:553

0x4008c141: vPortTaskWrapper at C:/work/esp-idf/components/freertos/port.c:143

Thanks,

**Comment by Saduf2019** [comment]
@YarivCol 
Please refer to [this issue comment](https://github.com/openthread/ot-esp32/issues/3#issuecomment-633198220) and let us know if it helps.

**Comment by YarivCol** [comment]
@Saduf2019 
Hi, The crash seems to be related to the Softmax layer, when we remove the layer everything works fine.
The Softmax fail asserting tfmicro/tensorflow/lite/kernels/internal/quantization_util.cc:96.

**Comment by mr-goldhands** [comment]
@YarivCol , cold you specify your Tensorflow **sources** SHA?
You can use the command: `git rev-parse --short HEAD`

**Comment by YarivCol** [comment]
@mr-goldhands 
Tensorflow sources SHA: b36436b087bd8e8701ef51718179037cccdfc26e

**Comment by YarivCol** [comment]
@mr-goldhands 
@rockyrhodes 
@jvishnuvardhan 
@Saduf2019 

Any updates?, I sent the commit SHA:  b36436b087bd8e8701ef51718179037cccdfc26e

**Comment by vikramdattu** [comment]
@YarivCol have you resolved the issue? Could you please check with latest master of TFLu?

**Comment by sushreebarsa** [comment]
@YarivCol  This is a micro related issue,Please post this in [micro repository](https://github.com/tensorflow/tflite-micro/issues)   and move this issue to close status..Thanks!
